staging/src/k8s.io/apiserver/pkg/server/genericapiserver.go

```diff
@@ -205,6 +205,9 @@ type DelegationTarget interface {
 
 	// NextDelegate returns the next delegationTarget in the chain of delegations
 	NextDelegate() DelegationTarget
+
+	// PrepareRun does post API installation setup steps. It calls recursively the same function of the delegates.
+	PrepareRun() preparedGenericAPIServer
```

Comment:
somewhat of a pre-existing issue (since PostStartHooks()/PreShutdownHooks() return unexported types), but declaring an interface method with an unexported return type makes it really hard for other interface implementers.